packages/adblocker/src/filters/network.ts

```diff
@@ -1771,7 +1795,12 @@ export default class NetworkFilter implements IFilter {
       if (this.filter !== undefined) {
         const skipLastToken = !this.isRightAnchor();
         const skipFirstToken = !this.isLeftAnchor();
-        tokenizeWithWildcardsInPlace(this.filter, skipFirstToken, skipLastToken, TOKENS_BUFFER);
+        tokenizeWithWildcardsInPlace(
+          this.isCaseSensitive ? this.filter : this.filter.toLowerCase(),
+          skipFirstToken,
+          skipLastToken,
+          TOKENS_BUFFER,
+        );
```

Comment:
This is required to comply with the reverse index filter discovery process. Tokens are "case-sensitive", and it will not be visible by `ReverseIndex.prototype.iterMatchingFilters` method. Basically, it will extract tokens from both filter and request then compare if the matching part is there. For an instance, a URL of `/foo/bar` will be tokenised into `foo` and `bar` then hashed. If we have a filter: `/foo/bar`, it will have same tokens. Then we compare those tokens and proceed to `NetworkFilter.prototype.match` if matched.